### DISCUSSION TITLE
How to get the autogenerated git branch url from vercel?

### DISCUSSION THREAD

**Discussion by morgs32** [vercel/community#1652] [Help] [answered]
### Summary

VERCEL_ENV is a system environment variable but it is the "unique hash" domain, not the "git" domain. Docs here: https://vercel.com/docs/concepts/projects/environment-variables

![image](https://user-images.githubusercontent.com/1364795/221935394-2982dd2b-7d36-43dc-8e34-5c2a7bd7122a.png)

**Comment by amyegan** [comment, marked as answer]
Hi Morgan! The `VERCEL_URL` value is the unique [generated url](https://vercel.com/docs/concepts/deployments/generated-urls#) specific to that deployment. But the [always up-to-date branch url](https://vercel.com/docs/concepts/deployments/generated-urls#url-with-git-branch) alias is predictable using multiple [url components](https://vercel.com/docs/concepts/deployments/generated-urls#url-components) with the format `<project-name>-git-<branch-name>-<scope-slug>.vercel.app`

**Reply by willTastyMakers** [reply]
I am sorry I don't understand your answer. The [URL with Git branch](https://vercel.com/docs/concepts/deployments/generated-urls#url-with-git-branch) link you described as the _always up-to-date branch_ url either does not work else I am just not perceiving the description correctly.

1.  I have a  Preview deploy to  branch called "dev" :  [  tms-nebula-cyfit76rz-wstreetertms.vercel.app](https://tms-nebula-cyfit76rz-wstreetertms.vercel.app/)

![Screenshot 2023-05-04 at 1 06 40 PM](https://user-images.githubusercontent.com/58872627/236318031-fc69dabb-08f7-4b66-8053-4150eee5b81b.png)

2. [URL with Git branch](https://vercel.com/docs/concepts/deployments/generated-urls#url-with-git-branch) describes the url structure for  _always up-to-date branch_ url   as 
      `<project-name>-git-<branch-name>-<scope-slug>.vercel.app`
      
      
3. My assessment is 
     project-name:  "tms-nebula" 
     branch-name: "dev"
     scope-slug: "wstreetertms"
     
     which suggest to me the url _always up-to-date branch_ url  =    https://tms-nebula-git-dev-wstreetertms.vercel.app   
     
     However, this has yet to work for any of my preview branches ( stage, dev)
     
     
If you could shed some light and what I am doing wrong or how I have misinterpreted these URL with Git Branch Instructions, it would be very helpful.

Thanks you

**Reply by amyegan** [reply]
Hi @willTastyMakers! I responded in https://github.com/vercel/community/discussions/2305
